### DISCUSSION TITLE
Is there any reason SyncBrowserHistory.js replaces spaces with plus sign?

### DISCUSSION THREAD

**Discussion by aguidrevitch** [livewire/livewire#4458] [7. Bugs (archived)]
In function    SyncBrowserHistory.js::updateState(method, url, response, component), line 150:
```
        try {
            if (decodeURI(url) != 'undefined') {
                url = decodeURI(url).replaceAll(' ', '+').replaceAll('\\', '%5C')
            }
```
For me it causes issue. My route looks like :
```
        Route::get('vendor_details/{vendor}', [VendorDetailsController::class, 'index'])
            ->name('vendor_details');
```

If vendor contains space eg 'vendor_details/SOME%20VENDOR', it works perfectly fine, eg while message contains fingerprint.path =  'vendor_details/SOME%20VENDOR'. However the window.history gets updated with vendor_details/SOME+VENDOR, and if I refresh the page, message's fingerprint.path becomes 'vendor_details/SOME+VENDOR' and I start to get redirected to the default route, even though in HttpConnectionHandler::applyPersistentMiddleware() $request->route() gets resolved correctly to VendorDetailsController.

Any other ideas why I get redirected to default route when + is in the URL? Maybe some checksum failure?

It also seems that plus sign is a reserved symbol according to https://datatracker.ietf.org/doc/html/rfc3986, and space should be represented via %20 instead. Furthermore, in chrome, spaces in url seems to be correctly changed to %20 when pushed through history.replaceState().

**Comment by joshhanley** [comment]
@aguidrevitch hmm not sure why that would be the case. That was changed in #3524, @nullthoughts if you get a chance, are you able to give us some insight into that change? Thanks!

**Reply by nullthoughts** [reply]
Thanks for copying me on this @joshhanley. 

@aguidrevitch: with #3524, to clean up query strings, the encoding was switched to RFC1738 which is proper for form data and therefore spaces are encoded as plus characters. An early idea was to have the encoding defined in the Livewire config, however the majority of query strings will be made up of form inputs and this value was hardcoded (and it was backwards compatible with older urls).

As to the specific example you provided: if you have a barebones example of a project with the routing setup, I'd be more than happy to take a look.

**Reply by joshhanley** [reply]
@nullthoughts thanks! Much appreciated 🙂

**Reply by Lupennat** [reply]
@nullthoughts  I opened a thread #4473 about an encoding issue a couple of days ago, it's something related to a different urldecode/urlencode match between javascript path vs php generated path.
I was working on a pull request but would like your opinion on it.
